### DISCUSSION TITLE
WatcherTower On kubernetes

### DISCUSSION THREAD

**Discussion by sidanaabhi** [containrrr/watchtower#1527] [General] [answered]
Is it possible to run this on kubernetes?
I tried deploying the image but got  level=error msg="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?" this error

**Comment by zoispag** [comment, marked as answer]
Nowadays there's no docker runtime at all for container runtime, so Watchtower cannot work, but even if it did, Kubernetes operations needs to happen by talking to the kube-api server, not a node socket, because you cannot know on which node a pod is scheduled.

I would suggest to look into either:

- [Argo CD Image Updater](https://argocd-image-updater.readthedocs.io/en/stable/) or
- [Keel](https://keel.sh/)
